### ISSUE TITLE
Use the new date and time picker

### ISSUE THREAD

**Issue by stevejburge** [publishpress/PublishPress-Future#10]
this is the clunky old box
<img width="341" alt="Screen Shot 2021-05-04 at 3 42 39 PM" src="https://user-images.githubusercontent.com/3868028/117060823-67965a00-acef-11eb-906e-0f1f40abdc33.png">

Let's move to the current WordPress default:
<img width="352" alt="Screen Shot 2021-05-04 at 3 41 54 PM" src="https://user-images.githubusercontent.com/3868028/117060871-78df6680-acef-11eb-8b9f-4e36a4fdd9a2.png">

**Comment by contactashish13** [comment]
@stevejburge This involves adding proper Gutenberg support and I'm almost about to finish this. I realize that unless we test this feature thoroughly by keeping in mind all the functionality of the classic meta box, the user may run into problems. For that reason, I thought it prudent to include the below section in settings:

![image](https://user-images.githubusercontent.com/12953439/125926150-39f221c0-98fc-405f-a716-9d6a61af0fef.png)

This is a temporary option and will allow users to disable the Gutenberg block and fall back to the current metabox similar to how it behaves right now. So in case of any glitches in our implementation, users won't get stuck.

Basically, `Show Gutenberg style box' will show

![image](https://user-images.githubusercontent.com/12953439/125926656-5ec1d2d0-72a0-4504-924c-1fcc4051fe8c.png)

and `Show Classic Editor style box` will show

![image](https://user-images.githubusercontent.com/12953439/125926935-27882efe-5ce6-4e06-80c3-f0acf966feb1.png)


Let me know if you think this option is not a good idea.

**Comment by stevejburge** [comment]
I like the way you think on this @contactashish13 

Yes, leaving the legacy option is 100% a good idea

**Comment by contactashish13** [comment]
@stevejburge PR: https://github.com/publishpress/PublishPress-Future/pull/65

In this I've take the first tentative step towards making the plugin fully OOP compliant by adding the new methods to classes. It does not affect any existing functionality.

**Comment by stevejburge** [comment]
@contactashish13 looking great! Nice work
